File: practice/src/listener.py
#!/usr/bin/env python3
#coding:utf-8

import rospy
from std_msgs.msg import Int64

def callbackfunc(msg):
    rospy.loginfo("%d\n",msg.data)

def subscriber():
    rospy.init_node('listener')
    rospy.Subscriber('counter', Int64, callbackfunc)
    rospy.spin()
    
if __name__=='__main__':
    try:
        subscriber()
    except rospy.ROSInterruptException:
        pass

File: practice/src/listener.cpp
#include "ros/ros.h"
#include "std_msgs/Int64.h"

void Callback(const std_msgs::Int64::ConstPtr& msg)
{
    std::cout<<"msg.data = "<<msg->data<<"\n";
}

int main(int argc, char **argv)
{
    ros::init(argc, argv, "listener");
    ros::NodeHandle nh;
    ros::Subscriber sub = nh.subscribe("counter",10,Callback);
    ros::Rate loop_rate(1);
    
    while(ros::ok())
    {
        ros::spinOnce();
        loop_rate.sleep();
    }
    return 0;
}
